### PR TITLE
 EDUCATOR-1104 Fix 500 error in course page.

### PR DESCRIPTION
**Pull request by attiyaIshaque**
### Description
This PR fixes that when the user is not enrolled in the course and credit state is None, it wouldn't raise the 500 error.

### How to Test?

**Production** 
- https://courses.edx.org/courses/course-v1:ASUx+SOC101x+2174C/course/

**Sandbox**
- https://proctoring-app.sandbox.edx.org/courses/course-v1:edx+PC101+2017/course/

### Tests
 - [x] Unit Test

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] @afzaledx 
- [x] @noraiz-anwar 

### Post-review
- [x] Rebase and squash commits

### CODE REVIEW

File: edx_proctoring/api.py [hunk 1]
```diff
@@ -1235,7 +1235,7 @@ def _check_eligibility_of_enrollment_mode(credit_state):
     # Also make an exception for the honor students to take the "practice exam" as a proctored exam.
     # For the rest of the enrollment modes, None is returned which shows the exam content
     # to the student rather than the proctoring prompt.
-    return credit_state['enrollment_mode'] == 'verified'
+    return credit_state and credit_state['enrollment_mode'] == 'verified'
```
Comment (noraiz-anwar) [inline edx_proctoring/api.py]: We can enhance readability of this by resolving `credit_state and credit_state['enrollment_mode'] == 'verified'` into a separate variable and then return it.
Something like:
``` python 
my_variable = credit_state and credit_state['enrollment_mode'] == 'verified'
return my_variable
```

File: edx_proctoring/tests/test_api.py [hunk 2]
```diff
@@ -1648,3 +1648,19 @@ def test_declined_prerequisites(self, content_id,
         self.assertEqual(len(results['failed_prerequisites']), expected_len_failed_prerequisites)
         self.assertEqual(len(results['pending_prerequisites']), expected_len_pending_prerequisites)
         self.assertEqual(len(results['declined_prerequisites']), expected_len_declined_prerequisites)
+
+    def test_summary_without_credit_state(self):
+        """
+        Test that status summary is None for unenrolled user
+        that does not have a corresponding credit state.
+        """
```
Comment (noraiz-anwar) [inline edx_proctoring/tests/test_api.py]: Sorry I didn't get it. What summary we are discussing here? and corresponding credit state?

**Comment by noraiz-anwar** [comment]
What are steps to observe your fix on sandbox?

File: edx_proctoring/api.py (hunk 1 above)
Comment (attiyaIshaque) [inline edx_proctoring/api.py]: To enhance readability its good but when `credit_state` is `None` its check `credit_state['enrollment_mode'] `also. and it raised this error
```
'NoneType' object has no attribute '__getitem__'
```
but in this case, if `credit_state` is `None` its returns in this point `return credit_state`
Make sense?

File: edx_proctoring/tests/test_api.py (hunk 2 above)
Comment (attiyaIshaque) [inline edx_proctoring/tests/test_api.py]: Can you please see this method [get_attempt_status_summary](https://github.com/edx/edx-proctoring/blob/master/edx_proctoring/api.py#L1452)? You will better understand.

**Comment by attiyaIshaque** [comment]
@noraiz-anwar Just observe course home page would not raise server error.

File: edx_proctoring/api.py (hunk 1 above)
Comment (noraiz-anwar) [inline edx_proctoring/api.py]: If `credit_state` is `None` it will not check `credit_state['enrollment_mode']` in 
``` python 
my_variable = credit_state and credit_state['enrollment_mode'] == 'verified'
```

File: edx_proctoring/tests/test_api.py (hunk 2 above)
Comment (noraiz-anwar) [inline edx_proctoring/tests/test_api.py]: We can reword this doc string and specify that we are going to check **attempt status summary**. Also we can specify about corresponding credit state, is this state related to enrollment? If yes, then we can simply say 'test that attempt status summary is None for users who are not enrolled'.

File: edx_proctoring/api.py (hunk 1 above)
Comment (afzaledx) [inline edx_proctoring/api.py]: Will the credit_state check be more appropriate here: https://github.com/edx/edx-proctoring/blob/2bee14371e8024e9bca0f7460c0df8538a147e59/edx_proctoring/api.py#L1494

Comment (noraiz-anwar) [inline edx_proctoring/api.py]: @afzaledx thanks for sharing this. 
@attiyaIshaque You can refactor `_check_eligibility_of_enrollment_mode` before using it. credit_state can be checked in this method before doing `credit_state['enrollment_mode'] == 'verified'`.

Comment (attiyaIshaque) [inline edx_proctoring/api.py]: @afzaledx Its better to check credit_state within the method `_check_eligibility_of_enrollment_mode`

File: edx_proctoring/tests/test_api.py (hunk 2 above)
Comment (attiyaIshaque) [inline edx_proctoring/tests/test_api.py]: @noraiz-anwar i will change the doc string :) thanks

File: edx_proctoring/api.py [hunk 3]
```diff
@@ -1235,7 +1235,9 @@ def _check_eligibility_of_enrollment_mode(credit_state):
     # Also make an exception for the honor students to take the "practice exam" as a proctored exam.
     # For the rest of the enrollment modes, None is returned which shows the exam content
     # to the student rather than the proctoring prompt.
-    return credit_state['enrollment_mode'] == 'verified'
+    if credit_state:
+        credit_state['enrollment_mode'] == 'verified'
+    return credit_state
```
Comment (noraiz-anwar) [inline edx_proctoring/api.py]: I think it is better to explicitly return `False` here. Caller of this function would be expecting a boolean. Some caller of this function may use this function like
``` python 
if _check_eligibility_of_enrollment_mode == True/False
```

**Comment by attiyaIshaque** [comment]
@noraiz-anwar I am sticking with my this code 
```
 return credit_state and credit_state['enrollment_mode'] == 'verified'
```
Now i am merging it.
